pkg/registry/apis/provisioning/export.go

```diff
@@ -57,49 +48,50 @@ func (c *exportConnector) Connect(
 	opts runtime.Object,
 	responder rest.Responder,
 ) (http.Handler, error) {
-	repo, err := c.repoGetter.GetHealthyRepository(ctx, name)
+	repo, err := c.repoGetter.GetRepository(ctx, name)
 	if err != nil {
 		return nil, err
 	}
-	ns := repo.Config().GetNamespace()
-	logger := logging.FromContext(ctx).With("logger", "export-connector", "repository", name, "namespace", ns)
-
-	// TODO: We need some way to filter what we export.
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		logger := logger.WithContext(ctx)
-		ctx := logging.Context(r.Context(), logger)
-
-		job, err := c.queue.Add(ctx, &provisioning.Job{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: ns,
-				Labels: map[string]string{
-					"repository": repo.Config().GetName(),
-				},
-			},
-			Spec: provisioning.JobSpec{
-				Action: provisioning.JobActionExport,
-			},
-		})
+		options := provisioning.ExportOptions{}
```

Comment:
I think we should stick to have `Export` as it is giving you the job id and then we have another endpoint we can use for any job. `/jobs/<id>/watch`. I think that would keep it cleaner and would be more reusable. 

I could see us using this mechanism to give status messages for `sync` as well :)

Reply:
I started with that... but then we are prioritizing async job for the thing we really want fast/sync info to the user that asked for it.

K8s can be ok for a few events -- but I am very nervous about depending on it for things where I would like to quickly stream info to the user (a progress bar!) without exercising all the overhead required for k8s 😬

Reply:
@MissingRoberto I updated the UI... messages are of course random -- but there is enough there that the design team can improve :)